sw/host/opentitanlib/src/transport/hyperdebug/i2c.rs

```diff
@@ -284,4 +370,131 @@ impl Bus for HyperdebugI2cBus {
         }
         Ok(())
     }
+
+    fn get_device_status(&self, timeout: Duration) -> Result<DeviceStatus> {
+        if !self.cmsis_encapsulation || self.mode.get() != Mode::Device {
+            return Err(TransportError::UnsupportedOperation.into());
+        }
+        let timeout_ms = timeout.as_millis();
+        let req = CmdGetDeviceStatus {
+            encapsulation_header: [Self::CMSIS_DAP_CUSTOM_COMMAND_I2C_DEVICE; 2],
+            port: self.bus_idx,
+            device_cmd: I2C_DEVICE_CMD_GET_DEVICE_STATUS,
+            timeout_ms: if timeout_ms > 65535 {
+                65535
+            } else {
+                timeout_ms as u16
+            },
+        };
+        self.usb_write_bulk(&req.as_bytes()[1..])?;
+
+        let mut resp = RspGetDeviceStatus::new();
+        let mut bytecount = 0;
+        while bytecount < 7 {
+            let read_count = self.usb_read_bulk_timeout(
+                &mut resp.as_bytes_mut()[1 + bytecount..][..64],
```

Comment:
Could you clarify what is happening here with the fixed offset of `1`?

In other bits of the i2c code, it looks like we want to send the encapsulation header byte if `cmsis_encapsulation`.  In these i2c-target-device structs, there are two bytes in the encapsulation header and it seems you're omitting one of them.

Reply:
Yes, I realize it is confusing.  I want one byte of CMSIS encapsulation header, but I was unsure how the `repr(C)` packing works, if I put a `u16` after a single `u8`.  In order to be sure not to get any padding between fields, I declared two `u8` in the beginning before the `u16`, and then I always skip the first when extracting the byte sequence to send.

Reply:
Alternatively, use [`zerocopy::Unalign`](https://docs.rs/zerocopy/latest/zerocopy/struct.Unalign.html)